src/common/extensionMessaging.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as Q from "q";
+import * as http from "http";
+
+/**
+ * Message server parameters.
+ */
+export let ServerParams = {
+    PORT: 8099,
```

Comment:
The port should be configurable, but that configuration will need to be communicated from the extension process to the debug adapter somehow. I suppose in the worst case we can ask the user to specify it in two locations: settings.json and launch.json?